src/transformers/models/encodec/modeling_encodec.py

```diff
@@ -0,0 +1,1664 @@
+        self.ratios = config.ratios
+        self.n_residual_layers = config.num_residual_layers
+        self.hop_length = np.prod(self.ratios)
+
+        act = getattr(nn, config.activation)
+        mult = int(2 ** len(self.ratios))
+        model = [
+            SConv1d(
+                config.dimension,
+                mult * config.num_filters,
+                config.kernel_size,
+                norm=config.norm,
+                norm_kwargs=config.norm_params,
+                causal=config.causal,
+                pad_mode=config.pad_mode
+            )
+        ]
+
+        if config.lstm:
+            model += [SLSTM(mult * config.num_filters, num_layers=config.lstm)]
+
+        # Upsample to raw audio scale
+        for i, ratio in enumerate(self.ratios):
+            # Add upsampling layers
+            model += [
+                act(**config.activation_params),
+                SConvTranspose1d(
+                    mult * config.num_filters,
+                    mult * config.num_filters // 2,
+                    kernel_size=ratio * 2,
+                    stride=ratio,
+                    norm=config.norm,
+                    norm_kwargs=config.norm_params,
+                    causal=config.causal,
+                    trim_right_ratio=config.trim_right_ratio,
+                )
+            ]
+            # Add residual layers
+            for j in range(config.num_residual_layers):
+                model += [
+                    SEANetResnetBlock(
+                        mult * config.num_filters // 2,
+                        kernel_sizes=[config.residual_kernel_size, 1],
+                        dilations=[config.dilation_base ** j, 1],
+                        activation=config.activation,
+                        activation_params=config.activation_params,
+                        norm=config.norm,
+                        norm_params=config.norm_params,
+                        causal=config.causal,
+                        pad_mode=config.pad_mode,
+                        compress=config.compress,
+                        true_skip=config.true_skip,
+                    )
+                ]
+            mult //= 2
+
+        # Add final layers
+        model += [
+            act(**config.activation_params),
+            SConv1d(
+                config.num_filters,
+                config.audio_channels,
+                config.last_kernel_size,
+                norm=config.norm,
+                norm_kwargs=config.norm_params,
+                causal=config.causal,
+                pad_mode=config.pad_mode
+            )
+        ]
+        # Add optional final activation to decoder (eg. tanh)
+        if config.final_activation is not None:
+            final_act = getattr(nn, config.final_activation)
+            final_activation_params = final_activation_params or {}
+            model += [
+                final_act(**final_activation_params)
+            ]
+        self.model = nn.Sequential(*model)
```

Comment:
ok to use sequential?

Reply:
Think it'd be slightly nicer / better fit transformers design to use a for loop here

Reply:
done